be/src/exprs/jit/jit_expr.cpp

```diff
@@ -121,16 +115,17 @@ StatusOr<ColumnPtr> JITExpr::evaluate_checked(starrocks::ExprContext* context, C
     return result_column;
 }
 
-void JITExpr::close(RuntimeState* state, ExprContext* context, FunctionContext::FunctionStateScope scope) {
-    auto* jit_engine = JITEngine::get_instance();
-    if (jit_engine->initialized()) {
-        auto status = jit_engine->remove_function(_expr->debug_string());
-        if (!status.ok()) {
-            LOG(WARNING) << "JIT: remove function failed, reason: " << status;
+// only unregister once
+JITExpr::~JITExpr() {
+    if (_is_prepared && _jit_function != nullptr) {
+        auto* jit_engine = JITEngine::get_instance();
+        if (jit_engine->initialized()) {
+            auto status = jit_engine->remove_function(_expr->debug_string());
+            if (!status.ok()) {
+                LOG(WARNING) << "JIT: remove function failed, reason: " << status;
+            }
         }
```

Comment:
Have you override JITExpr::clone method?  it seems required.

both runtime filter and pushdown filter needs invoke this clone method

Reply:
yes in .h file
```
    Expr* clone(ObjectPool* pool) const override { return JITExpr::create(pool, _expr); }
```